cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -2802,23 +2802,55 @@ void RegisterScalarArithmetic(FunctionRegistry* registry) {
   // ----------------------------------------------------------------------
   auto multiply = MakeArithmeticFunction<Multiply>("multiply", &mul_doc);
   AddDecimalBinaryKernels<Multiply>("multiply", multiply.get());
+
+  // Add multiply(duration, int64) -> duration
+  for (auto unit : TimeUnit::values()) {
+    auto exec = ArithmeticExecFromOp<ScalarBinaryEqualTypes, Multiply>(Type::DURATION);
+    DCHECK_OK(
+        multiply->AddKernel({duration(unit), int64()}, duration(unit), std::move(exec)));
```

Comment:
Is it worth also supporting `multiply(int64, duration)` or would commutativity be handled at an upper level? @lidavidm

Reply:
Hmm, I don't see why not. It could just use the same exec function.

Reply:
Done.